src/compiler/checker.ts

```diff
@@ -18193,6 +18193,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (objectType.flags & (TypeFlags.Any | TypeFlags.Never)) {
                 return objectType;
             }
+            if (indexType.flags & TypeFlags.Never) {

```

Comment:
XState failures would likely get fixed if you'd bring back the other piece of code and change this one to:
```suggestion
            if (indexType === neverType) {
```

XState relies on propagating `silentNeverType`. Previously `[silentNeverType][silentNeverType]` was evaluated to `silentNeverType` but with this change its `neverType`. Note that for this special never type (and there are some more of them in the compiler) you should go through the index signature lookup and not return `indexType` here directly (that just happens to match what indexing with `silentNeverType` evaluates to in our case).

Reply:
@Andarist can you expand further on how/why XState relies on propagating `silentNeverType`?

P.S. I implemented this check as you suggested :D

Reply:
P.P.S. Currently I've not bringed back the other check after the index signature one, do you really think it is needed or I can ask to rerun the tests?

Reply:
> @Andarist can you expand further on how/why XState relies on propagating silentNeverType?

To be honest, I have no idea. It's a magic internal type and I never had time to explore how this works.

> P.P.S. Currently I've not bringed back the other check after the index signature one, do you really think it is needed or I can ask to rerun the tests?

I'd check what happens now with `type Result = {}[never]`. It should be `never` but I have a feeling that it might be `any` right now or an error type (a special kind of `any`).

Reply:
I'm not sure what I was thinking about in the comment above. `type Result = {}[never]` is still `never` here because that `never` is just a regular `never` (so with the proposed change it just enters the branch that you have moved).